ras/ServiceReport.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM.
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+
+import os
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process, build
+from avocado.utils import archive
+from avocado.utils.software_manager import SoftwareManager
+
+
+class ServiceReport(Test):
+
+    """
+    ServiceReport is a tool to validate and repair system configuration for
+    specific purposes.Initially envisaged to help setup systems for correct
+    First Failure Data Capture (FFDC),it has now morphed into a plugin based
+    framework which can do more than just FFDC validation.
+
+    :avocado: tags=privileged
+    """
+
+    fail_cmd = list()
+
+    def run_cmd(self, cmd):
+        if process.system(cmd, ignore_status=True, sudo=True, shell=True):
+            self.fail_cmd.append(cmd)
+
+    def error_check(self):
+        if len(self.fail_cmd):
+            for cmd in self.fail_cmd:
+                self.log.info("Failed command: %s" % cmd)
+            self.fail("ServiceReport: Failed commands are: %s" % self.fail_cmd)
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.plugin = self.params.get('plugin', default='kdump')
+        for package in ['make', 'gcc']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel("Fail to install %s required for this"
+                            " test." % package)
+        tarball = self.fetch_asset('ServiceReport.zip', locations=[
+                                   'https://github.com/linux-ras/ServiceReport'
+                                   '/archive/master.zip'], expire='7d')
+        archive.extract(tarball, self.workdir)
+        self.sourcedir = os.path.join(self.workdir, 'ServiceReport-master')
+        build.make(self.sourcedir)
+
+    def test(self):
+        os.chdir(self.sourcedir)
+        cmd_list = ['', '-l', '-v', '-r', '-p %s' % self.plugin]
```

Comment:
Why can't this be multiplexed ?

Reply:
@narasimhan-v no don't think these are basic option need to be checked, anyway, this is basic commit will modify about our need and have more error handling

Reply:
@PraveenPenguin What I meant was, why use for loop when avocado does it for us. We can even get per parameter result.

Reply:
@narasimhan-v ok but still I  feel should be part of the test itself as error processing should be different as based on return output that will be targeted on enhancement as right now it is same for all the cases and don't want to run setup every time so coded this way i know we can skip setUP as well but I  need to modify logic based on enhancement how this sound you?

Reply:
@PraveenPenguin Sure, if you have planned enhancements then this might be fine.
Suggestion: please explore if error handling also can be provided as input via yaml.